### ISSUE TITLE
PNG 16bpp Support

### ISSUE THREAD

**Issue by PeterMerkert** [inalogic/pico-pixel-public#23]
Hi, just have a test with a 16f single channel (== grayscale) PNG and wanted to investigate it with the super cool pico pixel. However, you apparently do not support it as PP tells me when it is opened up it is BRG8 and it is completly black.

I got the resource from a friend and do not know how to create 16f PNGs in Photoshop. 

The PNG spec is also telling me that 16f is possible: http://www.libpng.org/pub/png/spec/1.2/PNG-Chunks.html

**Comment by PeterMerkert** [comment]
Ah, just checked Photoshop again: You can just create a RGB16f image and save it. It will preserve the format. Alpha channel will be dropped - in case there was any.

**Comment by jaytaoko** [comment]
I am not sure that PNG supports  a single channel 16 bits floating-point. However it does supports single channel 16 bits integer. I looked online to try and find a png file with format 16F but I couldn't find any. There is a mention here http://www.libpng.org/pub/png/spec/1.1/pngext-1.1.0-pdg.html#DR.Floating-point-values
`The core of PNG does not use floating-point numbers anywhere; it uses integers or, where applicable, fixed-point fractional values.`
Can Photoshop save textures as RGB16F PNG? Or do you need a plugin for that? Would it be possible for you to share the png file that is not working in PicoPixel?
Thanks

**Comment by PeterMerkert** [comment]
Yes, you are right @jaytaoko , it is 16 bit integer values. I am sorry for the wrong information. I though the texture someone provided to me was in Half Float, but it is not. It is in shorts (so, 16bit integers).

I will adjust my request then to: Can Pico Pixel implement support for 16i PNG format? :)

**Comment by jaytaoko** [comment]
Sure, I will add support for it. Thank you for your feedback.

**Comment by jaytaoko** [comment]
I have verified the support for 16 bit png images in PicoPixel. Support was already there. However, the image format wasn't exposing the true native format of the file. I have fixed that issue. Now PicoPixel can show the native format of an image to be gray2 (2bits), gray4, gray8 or gray16.

![picopixel-gray16](https://cloud.githubusercontent.com/assets/1514772/20460601/d1310a06-aeb5-11e6-9d25-ab88bf5a4f26.png)

If you still think there is an issue with the image you have been using, I would appreciate if you can send it to me for further inspection.
Regards
